### ISSUE TITLE
Any plans on migrating to Manifest V3?

### ISSUE THREAD

**Issue by m4heshd** [sniklaus/youtube-watchmarker#129]
First of all, thank you @sniklaus so much for this unbelievably useful extension. Been using this for years.

Seems like this is one of my favorite extensions that I cannot live without which will soon be devoured by Google's manifestocalypse. I'm not huge on developing extensions but looking at this with my limited knowledge, it should be possible to migrate to v3 without compromising on functionality right?

I really hope this extension can survive what's to come.

**Comment by sniklaus** [comment]
Thank you for the kind words. Yes, it is possible to convert the extension. Right now I have not invested any time into this yet though since I am waiting to see what will happen. I am still hopeful that Google will let this all go once they realize how few extensions will be left if the force v3 upon everyone, fingers crossed.

**Comment by sniklaus** [comment]
Oh boy, they are really following through with this. Is anyone also already facing this?

**Comment by BrianJ20** [comment]
Can no longer be installed via 

https://chromewebstore.google.com/detail/watchmarker-for-youtube/pfkkfbfdhomeagojoahjmkojeeepcolc

At least for me.

**Comment by yarrmateys** [comment]
naturally they're following through. this is google, a corporation hell bent on controlling what you can do with the internet, extension compatibility were never their concern. "if you don't like it, you can switch to an alternative browser, almost all of which are based on chrome" is probably what they think.

**Comment by serossi** [comment]
then again youtube is unwatchable without adblock but that will not work with v3 either
so this should be now a firefox and brave only extension xD

**Comment by BrianJ20** [comment]
> then again youtube is unwatchable without adblock but that will not work with v3 either so this should be now a firefox and brave only extension xD

https://chromewebstore.google.com/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh
v3 adblock works fine

**Comment by serossi** [comment]
> > then again youtube is unwatchable without adblock but that will not work with v3 either so this should be now a firefox and brave only extension xD
> 
> https://chromewebstore.google.com/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh v3 adblock works fine

no it doesnt, the "lite" in this context means basically useless
https://ublockorigin.com/
the statement to v3 is on their mainpage

**Comment by BrianJ20** [comment]
> > > then again youtube is unwatchable without adblock but that will not work with v3 either so this should be now a firefox and brave only extension xD
> > 
> > 
> > https://chromewebstore.google.com/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh v3 adblock works fine
> 
> no it doesnt, the "lite" in this context means basically useless https://ublockorigin.com/ the statement to v3 is on their mainpage

That statement is regarding V3 and uBlock Origin. Not uBlock Origin Lite.  It states right below the statement that uBlock Origin Lite is a solution to uBlock Origin being broken.

But words don't matter. All that matters is, i've used it for 2 months and haven't seen an ad on youtube yet.

But thanks for your input.

**Comment by serossi** [comment]
> > > > then again youtube is unwatchable without adblock but that will not work with v3 either so this should be now a firefox and brave only extension xD
> > > 
> > > 
> > > https://chromewebstore.google.com/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh v3 adblock works fine
> > 
> > 
> > no it doesnt, the "lite" in this context means basically useless https://ublockorigin.com/ the statement to v3 is on their mainpage
> 
> That statement is regarding V3 and uBlock Origin. Not uBlock Origin Lite. It states right below the statement that uBlock Origin Lite is a solution to uBlock Origin being broken.
> 
> But words don't matter. All that matters is, i've used it for 2 months and haven't seen an ad on youtube yet.
> 
> But thanks for your input.

seriously ? how condesneding of you with your non existent knowlege

ublock lite is a cutdown version of ublock statement stands

no thanks for your input

**Comment by AntonZhuchkovXOR** [comment]
Today Chrome switched Watchmarker off because of «extension is no longer supported». I could switch it on in the extension settings but in the Chrome Store it says that extension is not longer available.

**Comment by johan456789** [comment]
I'm working on it #148, starting with refactoring in manifest v2, potentially adding some tests before we move on to migrating to manifest v3

**Comment by shanec42** [comment]
Hey Sniklaus, 

If you're not able (because of effort, time, money, motivation*, etc) to bring this extension up to v3 status to make the big G happy, could I ask that you make an easily downloadable package for side loading via "Local unpacked?"  I had to jump through some hoops to make the git clone completely work.  I documented it here https://chromewebstore.google.com/support/4e8bf21a-0ce8-4075-883a-d4c94dbf53a1

Mostly, it was the content directory that was missing files that I was able to grab from the FireFox install.  Yeah, Chrome's not happy about the manifest file. {shrug}  It still works.  And that's all I care about.

Personally, I wouldn't worry about providing instructions on how to side-load the extension. People who are technically capable of doing it can find those instructions elsewhere.  And people who aren't technically capable probably shouldn't be doing it to begin with.

Thanks for this great extension!

Shane

*: Hey, I understand.  Your time, your interest, your motivation.

**Comment by johan456789** [comment]
migration to manifest v3 is a must now if you want to keep using it after june 2025 ([source](https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline#june_2025_chrome_mv2_deprecation_enterprise_rollout)). Before that you can follow the instructions on [reddit](https://www.reddit.com/r/uBlockOrigin/comments/1itw1bz/end_of_support_for_ubo_on_chrome_chromium/) to re-enable manifest v2 support.

For those who didn't have the extension downloaded from chrome web store already, you can grab the zip file at my [fork](https://github.com/johan456789/youtube-watchmarker/releases/tag/synced-w-published), unzip it and load unpacked to chrome (developer mode needs to be turned on.)

**Comment by sniklaus** [comment]
> could I ask that you make an easily downloadable package for side loading via "Local unpacked?"

Sure thing, will provide them here moving forward: https://github.com/sniklaus/youtube-watchmarker/releases

**Comment by rossc719g** [comment]
> The database of the original extension appears to be hashed, so your raw JSON import feature is not functional.

The old database is just base64 encoded. (I'm not clear why...)

If you decode that, it's just json.

**Comment by rlnt** [comment]
I tried the latest commit, but it crashes my browser when I attempt to import my old database. I assume it runs out of memory because the file is too large. It's 5 MB.

I worked around it by manually decoding the base64, and then I used [json splitter](https://github.com/jhsu98/json-splitter) with some minor tweaks (made the file read operation use UTF-8) to split the file into parts of 0.5 MB max. I was then able to import 11 parts one by one and now I have a database of 38.716 entries.

**Comment by rossc719g** [comment]
Thanks for adopting this extension!

Since you asked.. I might have one for you:

I haven't updated the unpacked version since I installed it.. but as I have been using it, I noticed that it doesn't seem to be behaving like it used to.

In particular, if I watch a video on machine A, it seems that I will then see the same videos unwatched a few days later on machine B.

I know there is no shared storage between A and B... but the older version seemed to do some sort of periodic dl of the watched history from YT to keep up with videos watched on other devices.   I.e., B would occasionally check in with YouTube's history list and add the videos I watched on A to the db on B. 

Is that background-watch-history feature still in the new version?  It doesn't seem to be happening automatically anymore.  
If I go into the options and press the button to do it manually, it seems to work fine and all the videos I watched on A are added on B.

**Comment by m4heshd** [comment]
Just lost my Manifest v2 privileges and missed the opportunity to backup the watch history. Any chance I could export the extension's local storage now after being disabled?

I did manage to find a backup from 2 months back, which has almost 60k watched entries, but is missing more than two months of watch history.

Thank you so much for your work @widike. I just started using your fork, and it looks slick. Functionality is top-notch as well. To add to that, the code looks very clean, modularized, and maintainable now. Hoping to contribute in the future.

The database import went well. Hope it picks up the rest from my browser history since the API query is limited to 200 records.

**Comment by m4heshd** [comment]
@widike I'm still using your fork of this extension because of the better management UI and stuff. You still maintain it?

**Comment by widike** [comment]
@m4heshd Yep! I still use it myself, so I'll definitely keep fixing things as they break. I push updates to the repo every now and then, but I usually don't bother cutting formal releases and I removed Firefox support. I went ahead and made a new release just now, though!

**Comment by m4heshd** [comment]
@widike That is such a relief. Please do continue to publish releases. Otherwise, we users don't get notified of new updates. Thank you so much for continuing to maintain it.

In the rare chance that you aren't able to maintain your fork anymore, is it still possible to migrate a backup from your version to this version? I'm terrified of losing my watch history once again.

**Comment by widike** [comment]
Yes, the databases are pretty much the same. Check this issue for the difference: https://github.com/widike/youtube-watchmarker/issues/5
